### DISCUSSION TITLE
HUD score rolls up from 0 rather than starting from most current values when spectating an ongoing multiplayer lobby

### DISCUSSION THREAD

**Discussion by GDcheerios** [ppy/osu#22533] [Q&A]
![image](https://user-images.githubusercontent.com/54640414/128597970-c7196a16-21bb-4e0f-b585-d5adbb3109c1.png)
I have score display mode set to classic and it appears as standardized while spectating in multiplayer

**Comment by bdach** [comment]
No, that assertion sounds incorrect. If the score was standardised it should never exceed one million aside from bonuses (which the score totals already exceeded on the attached screenshot).

I'd wager that it is more likely that the scores are rolling up to their actual value on the HUD. Score values appear instantaneously on the spectator HUD and roll up to the target value on the actual player HUD. So if there's anything to fix here it's probably that. Did you spectate said game when it was already going on?

**Comment by GDcheerios** [comment]
no I was using classic
![image](https://user-images.githubusercontent.com/54640414/128598299-d68689b3-b006-4320-9734-6a708ccec629.png)

**Comment by bdach** [comment]
...I can read, thanks.

I don't understand how you came to the conclusion that it "appears as standardised" from the screenshot attached. The scores, as I already said, are above 1 million on the screenshot. So I would say that they are *classic* scores and *not* standardised.

**Comment by GDcheerios** [comment]
sorry I didn't mean to come off as rude, and also I am very sorry for wasting your time I just realized that they were above million I saw that the top of the screenshot was lower than the other scores and immediately thought that they were not classic.

"Did you spectate said game when it was already going on?"

to answer your question yes.

**Comment by bdach** [comment]
Then as stated above, maybe the scores should instantaneously get the current value rather than rolling up from 0 when entering spectator mode for an ongoing match. Going to wait for more opinions on this first, before anything else, though.

**Comment by peppy** [comment]
I'm having trouble following this conversation - did the original poster actually mistake the animation of the score display for incorrect values?

**Comment by bdach** [comment]
That is what I believe to have happened, yes.

**Comment by peppy** [comment]
I'm not sure I follow though. The rolling period is only 1 second, so this would have been showing the correct score within a second. Is there something more to this?

**Comment by Theighlin** [comment]
Tried on the latest release of the game on an ongoing lobby at the time of writing.
If i'm understanding this issue correctly then this seems to be fixed. You can see that the scores are higher on reentering spectator. Works with both standard and classic scoring display.

https://user-images.githubusercontent.com/53081750/214954132-9db5b781-e1c6-4d58-90fd-7413df414db5.mp4

**Comment by bdach** [comment]
I don't think this technically closes the issue as stated, because the issue as stated is about the discrepancy between the leaderboard on the left having actual user scores, and the player instances having zeroes until they catch up to gameplay and finish rolling. That said, I'm also not sure how valid this report is; I don't really see the rolling as that big of a problem now. Dunno.

**Comment by Theighlin** [comment]
So it's just the visual of it starting from zero together with the rolling. Sorry, I misunderstood the issue.

While it may technically be an issue i don't think it's really important. From the above video the combo also seems to be 0 while it's waiting for the spectator data, though that doesn't roll up, and accuracy is 100% on some clients and 0% on others (one is at 90%), some pp counters are at 0 and others not. They're all kinda behaving differently in that frozen instant.

_If this is a problem_ (personally it's not), I would solve this by uniforming the HUD values, either they all start at 0 or they all get the correct data in that moment.

**Comment by peppy** [comment]
I'm just gonna migrate this to a discussion. Not worth fixing unless we get a lot of people annoyed by it.
